debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -131,7 +165,39 @@ public static long contiguousHeaderSize()
 		return J9IndexableObjectContiguous.SIZEOF;
 	}
 
+	/**
+	 * Get the header size of the contiguous indexable object with the dataAddress header field.
+	 */
+	public static long contiguousHeaderSizeWithDataAddress() {
+		if (mixedReferenceMode) {
+			if (compressObjectReferences) {
+				return J9IndexableObjectWithDataAddressContiguousCompressed.SIZEOF;
+			}
+			return J9IndexableObjectWithDataAddressContiguousFull.SIZEOF;
+		}
+		return J9IndexableObjectWithDataAddressContiguous.SIZEOF;
+	}
+
+	/**
+	 * Get the header size of the discontiguous indexable object.
+	 */
 	public static long discontiguousHeaderSize()
+	{
+		try {
+			if (!J9BuildFlags.env_data64 || J9RASHelper.getVM(DataType.getJ9RASPointer()).isIndexableDataAddrPresent().isZero()) {
+				return discontiguousHeaderSizeLegacy();
+			} else {
+				return discontiguousHeaderSizeWithDataAddress();
+			}
```

Comment:
Again, suggest this be inverted.

Reply:
https://github.com/eclipse-openj9/openj9/pull/16096#discussion_r1119110475